### DISCUSSION TITLE
Saving multiple videos within an experimental session

### DISCUSSION THREAD

**Discussion by MForsthofer** [bonsai-rx/bonsai#1593] [Q&A] [answered]
Hi, i'm currently working on a workflow for closed-loop stimulation in tadpoles. Currently, only when a specific criterion is met, a video will be recorded for 10 seconds, and then the workflow will wait for the next trigger. However, i'm having trouble naming the files. Ideally i'd want them to be the date and time when the session was started, followed by a number indicating the stimulus repetition (so e.g. tadpole_2023-10-18_16-19-44_1. attached is the workflow i'm using to try doing this. in this iteration, it seems the filename isnt updated fast enough, so what i end up with is the date and time of the previous recording. If i clear that, then the workflow crashes due to an invalid filename.  Screenshot one is how i get date and time and then broadcast them as subjects (basically i get a timestamp of the start of the recording, add that to the pathname, and add an .avi in the expression transform). 
![image](https://github.com/bonsai-rx/bonsai/assets/52410254/834e5ebe-bf44-4fc4-8f3a-f45e4a725a00)

Screenshot two is how i then take e.g. tadpole_filename, and slot it into the videowriter node. #
![image](https://github.com/bonsai-rx/bonsai/assets/52410254/1110e27f-f0a3-476c-b62a-aff6b37ad99d)

Any idea how to improve the workflow to solve (this specific) issue?

**Comment by bruno-f-cruz** [comment]
Attach your workflow please

**Reply by MForsthofer** [reply]
Here's the whole workflow:
https://github.com/MForsthofer/swim_tracking

**Reply by MForsthofer** [reply]
Fair warning, it's probably not up to workflow design standards yet

**Comment by MForsthofer** [comment, marked as answer]
Using this workflow for saving videos rather than what i did solved my problem, (as well as switching to BehaviorSubject rather than PublishSubject to broadcast the filenames)
https://github.com/orgs/bonsai-rx/discussions/1597
